File: tests-ts/pages/BasePage.ts
import { Page } from '@playwright/test';

export class BasePage {
  constructor(protected page: Page) {}

  async goto(url: string): Promise<void> {
    await this.page.goto(url);
  }

  getUrl(): string {
    return this.page.url();
  }

  async waitForLoadState(state: 'load' | 'domcontentloaded' | 'networkidle' = 'networkidle'): Promise<void> {
    await this.page.waitForLoadState(state);
  }
}

File: tests-Python/pages/base_page.py
"""
Базовий клас для всіх Page Objects.
Містить загальні методи для роботи зі сторінками.
"""
from playwright.sync_api import Page, expect


class BasePage:
    """Базовий клас для всіх сторінок"""
    
    def __init__(self, page: Page):
        """
        Ініціалізація базової сторінки
        
        Args:
            page: Екземпляр Playwright Page
        """
        self.page = page
    
    def goto(self, url: str):
        """Перехід на сторінку"""
        self.page.goto(url)
    
    def get_url(self) -> str:
        """Отримати поточний URL"""
        return self.page.url
    
    def wait_for_load_state(self, state: str = "networkidle"):
        """Чекати завантаження сторінки"""
        self.page.wait_for_load_state(state)
    
    def take_screenshot(self, path: str):
        """Зробити скріншот сторінки"""
        self.page.screenshot(path=path)
